app/utils/url/path.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export const ID_PATH_PATTERN = '[a-z0-9]{26}';
+export const PLUGIN_ID_PATH_PATTERN = '^[a-zA-Z0-9-_.]+$';
+
+// This should cover:
+// - Team name (lowercase english characters, numbers or -)
+// - Two ids separated by __ (userID__userID)
+export const TEAM_NAME_PATH_PATTERN = '[a-z0-9\\-_]+';
```

Comment:
The problem I see here is with non latin characters like japanese korean etc...

Reply:
Only English characters are allowed. Maybe you're thinking of the display name? https://github.com/mattermost/mattermost/blob/51715d5ed695efea3aa3ff86a07951ca0bf6177a/server/public/model/team.go#L209-L219